sdk/identity/Azure.Identity/src/AzureIdentityEventSource.cs

```diff
@@ -50,6 +52,43 @@ internal sealed class AzureIdentityEventSource : AzureEventSource
 
         public static AzureIdentityEventSource Singleton { get; } = new AzureIdentityEventSource();
 
+        public bool IsEnabled(EventLogLevel eventLogLevel)
+        {
+            return eventLogLevel switch
+            {
+                EventLogLevel.Critical => IsEnabled(EventLevel.Critical, EventKeywords.All),
+                EventLogLevel.Error => IsEnabled(EventLevel.Error, EventKeywords.All),
+                EventLogLevel.Warning => IsEnabled(EventLevel.Warning, EventKeywords.All),
+                EventLogLevel.Informational => IsEnabled(EventLevel.Informational, EventKeywords.All),
+                EventLogLevel.Verbose => IsEnabled(EventLevel.Verbose, EventKeywords.All),
+                EventLogLevel.LogAlways => IsEnabled(EventLevel.LogAlways, EventKeywords.All),
+                _ => false,
+            };
+        }
+
+        public void Log(LogEntry entry)
+        {
+            switch (entry.EventLogLevel)
+            {
+                case EventLogLevel.Critical when IsEnabled(EventLevel.Critical, EventKeywords.All):
+                    LogMsalCritical(entry.Message);
+                    break;
+                case EventLogLevel.Error when IsEnabled(EventLevel.Error, EventKeywords.All):
+                    LogMsalError(entry.Message);
+                    break;
+                case EventLogLevel.Warning when IsEnabled(EventLevel.Warning, EventKeywords.All):
+                    LogMsalWarning(entry.Message);
+                    break;
+                case EventLogLevel.Informational when IsEnabled(EventLevel.Informational, EventKeywords.All):
+                    LogMsalInformational(entry.Message);
+                    break;
+                case EventLogLevel.Verbose when IsEnabled(EventLevel.Verbose, EventKeywords.All):
+                case EventLogLevel.LogAlways when IsEnabled(EventLevel.Verbose, EventKeywords.All):
```

Comment:
LogAlways should be the same as Info. See [Identity Web](https://github.com/AzureAD/microsoft-identity-web/blob/60821501ac7585ba68633cba3c7982287fa92181/src/Microsoft.Identity.Web.TokenAcquisition/TokenAcquisition.cs#L1085-L1086). It prints more low level / basic logs than informational.
 
`case EventLogLevel.LogAlways when IsEnabled(EventLevel.Informational, EventKeywords.All):`

Reply:
If this is true, why is there a discrete log level defined for LogAlways?

Reply:
> Good catch, but why wouldn't we map to the discrete `EventLevel.LogAlways`?

Yes, I think that'd be better.